.github/workflows/thread-sanitizer.yaml

```
@@ -0,0 +1,163 @@
+name: Thread Sanitizer analysis
+on:
+  push:
```

Set this so we can launch it manually at any time
```suggestion
  workflow_dispatch:
  push:
```